modules/ibm-z-rhel-kvm-host-recommendations.adoc

```diff
@@ -0,0 +1,181 @@
+   ...
+</domain>
+----
+<1> The number of I/O threads.
+<2> The driver element of the disk device.
+
+Threads can increase the performance of I/O operations for disk devices, but they also use memory and CPU resources. You can configure multiple devices to use the same thread. Which mapping of threads to devices leads to the best results depends on the available resources and the workload.
+
+Start with a small number of I/O threads. Often, a single I/O thread for all disk devices is sufficient. Do not configure more threads than the number of virtual CPUs, and do not configure idle threads.
+
+You can use the `virsh iothreadadd` command to add I/O threads with specific thread IDs to a running virtual server.
+
+== Avoid virtual SCSI devices
+
+Configure virtual SCSI devices only if you need to address the device through SCSI-specific interfaces. Configure disk space as virtual block devices rather than virtual SCSI devices, regardless of the backing on the host.
+
+However, you might need SCSI-specific interfaces for:
+
+* A LUN for a SCSI-attached tape drive on the host.
+
+* A DVD ISO file on the host file system that is mounted on a virtual DVD drive.
+
+== Configure guest caching for disk
+
+Configure your disk devices to do caching by the guest and not by the host.
+
+Ensure that the driver element of the disk device includes the `cache="none"` and `io="native"` parameters.
+
+[source,terminal]
+----
+<disk type="block" device="disk">
+    <driver name="qemu" type="raw" cache="none" io="native" iothread="1"/>
+...
+</disk>
+----
+
+== Exclude the memory balloon device
+
+Unless you need a dynamic memory size, do not define a memory balloon device and ensure that libvirt does not create one for you. Include the `memballon` parameter as a child of the devices element in your domain configuration XML file.
+
+. Check the list of active profiles:
++
+[source,terminal]
+----
+<memballoon model="none"/>
+----
+
+== Tune the CPU migration algorithm of the host scheduler
+
+[IMPORTANT]
+====
+Do not change the scheduler settings unless you are an expert who understands the implications. Do not apply changes to production systems without testing them and confirming that they have the intended effect.
+====
+
+The `kernel.sched_migration_cost_ns` parameter specifies a time interval in nanoseconds. After the last execution of a task, the CPU cache is considered to have useful content until this interval expires. Increasing this interval results in fewer task migrations. The default value is 500000 ns.
+
+If the CPU idle time is higher than expected when there are runnable processes, try reducing this interval. If tasks bounce between CPUs or nodes too often, try increasing it.
+
+To dynamically set the interval to 60000 ns, enter the following command:
+
+[source,terminal]
+----
+# sysctl kernel.sched_migration_cost_ns=60000
+----
+
+To persistently change the value to 60000 ns, add the following entry to /etc/sysctl.conf:
+
+[source,terminal]
```

Comment:
We use [rouge](https://github.com/rouge-ruby/rouge) for our syntax highlighting. When looking at the available source formatting list (`rougify list`), there is an alias for `.conf` files:

`[source,config]`

You could use that for better syntax highlighting

Reply:
@codyhoag From where do I get that "rougify list" couldn't find it in the link?

Reply:
@SNiemann15 You can run `gem install rouge` to install it. Then `rougify list` should work, IIRC.